plugins/woocommerce/src/Internal/Utilities/WebhookUtil.php

```diff
@@ -0,0 +1,136 @@
+	 *
+	 * @param int      $old_user_id ID of the deleted user.
+	 * @param int|null $new_user_id ID of the user to reassign existing data to, or null if no re-assignment is requested.
+	 *
+	 * @return void
+	 * @since 7.8.0
+	 */
+	private function reassign_webhooks_to_new_user_id( int $old_user_id, ?int $new_user_id ): void {
+		$webhook_ids = $this->get_webhook_ids_for_user( $old_user_id );
+
+		foreach ( $webhook_ids as $webhook_id ) {
+			$webhook = new \WC_Webhook( $webhook_id );
+			$webhook->set_user_id( $new_user_id ?? 0 );
+			$webhook->save();
+		}
+	}
+
+	/**
+	 * When users are about to be deleted show an informative text if they have webhooks assigned.
+	 *
+	 * @param \WP_User $current_user The current logged in user.
+	 * @param array    $userids Array with the ids of the users that are about to be deleted.
+	 * @return void
+	 * @since 7.8.0
+	 */
+	private function maybe_render_user_with_webhooks_warning( \WP_User $current_user, array $userids ): void {
+		global $wpdb;
+
+		$at_least_one_user_with_webhooks = false;
+
+		foreach ( $userids as $user_id ) {
+			$webhook_ids = $this->get_webhook_ids_for_user( $user_id );
+			if ( empty( $webhook_ids ) ) {
+				continue;
+			}
+
+			$at_least_one_user_with_webhooks = true;
+
+			$user_data      = get_userdata( $user_id );
+			$user_login     = false === $user_data ? '' : $user_data->user_login;
+			$webhooks_count = count( $webhook_ids );
+
+			$text = sprintf(
+				/* translators: 1 = user id, 2 = user login, 3 = webhooks count */
+				_nx(
+					'User #%1$s %2$s has created %3$d webhook.',
+					'User #%1$s %2$s has created %3$d webhooks.',
+					$webhooks_count,
+					'user webhook count',
+					'woocommerce'
+				),
+				$user_id,
+				$user_login,
+				$webhooks_count
+			);
+
+			echo '<p>' . esc_html( $text ) . '</p>';
+		}
+
+		if ( ! $at_least_one_user_with_webhooks ) {
+			return;
+		}
+
+		$webhooks_settings_url = esc_url_raw( admin_url( 'admin.php?page=wc-settings&tab=advanced&section=webhooks' ) );
+
+		// This block of code is copied from WordPress' users.php.
+		// phpcs:disable WooCommerce.Commenting.CommentHooks, WordPress.DB.PreparedSQL.NotPrepared
+		$users_have_content = (bool) apply_filters( 'users_have_additional_content', false, $userids );
+		if ( ! $users_have_content ) {
+			if ( $wpdb->get_var( "SELECT ID FROM {$wpdb->posts} WHERE post_author IN( " . implode( ',', $userids ) . ' ) LIMIT 1' ) ) {
+				$users_have_content = true;
+			} elseif ( $wpdb->get_var( "SELECT link_id FROM {$wpdb->links} WHERE link_owner IN( " . implode( ',', $userids ) . ' ) LIMIT 1' ) ) {
+				$users_have_content = true;
+			}
+		}
+		// phpcs:enable WooCommerce.Commenting.CommentHooks, WordPress.DB.PreparedSQL.NotPrepared
+
+		if ( $users_have_content ) {
+			$text = __( 'If the "Delete all content" option is selected, the affected webhooks will <b>not</b> be deleted and will be attributed to user id 0.<br/>', 'woocommerce' );
```

Comment:
```suggestion
			$text = __( 'If the "Delete all content" option is selected, the affected WooCommerce webhooks will <b>not</b> be deleted and will be attributed to user id 0.<br/>', 'woocommerce' );
```

- Explains to the user that these are WooCommerce webhooks, as the user doing the deletion may not be working in a WooCommerce context.